### PR TITLE
optimize lro/rpc-legacy test case

### PR DESCRIPTION
**Pull request by msyyc**
# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/azure/core/lro/rpc-legacy/main.tsp
Comment (msyyc) [inline packages/cadl-ranch-specs/http/azure/core/lro/rpc-legacy/main.tsp:146]: Template can't describe whether the initial response has body. Here is the reference: https://azure.github.io/typespec-azure/docs/howtos/Azure%20Core/long-running-operations#example-3---link-to-statusmonitor-in-the-location-header

**Review by tadelesh — APPROVED** [review]

**Review by weidongxu-microsoft — COMMENTED** [review]
unblock

Please get approval from .NET, they seems having the test.

**Comment by msyyc** [comment]
@chunyu3 @pshao25 @archerzz Please help review

**Review by archerzz — APPROVED** [review]
.NET is fine.

**Pull request merged**
